commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpClientFactory.java

```diff
@@ -125,9 +125,9 @@ public static Session createConnection(final String hostname, final int port, fi
                 final SftpFileSystemConfigBuilder.ProxyType proxyType = builder.getProxyType(fileSystemOptions);
```

Comment:
Why not just pull out the proxy user and password the same way the proxy host is pulled out, instead of mixing parameters + config builder in the createXXX functions?

Reply:
The convention from the STREAM type seemed to be to pass the Opts and Builder, so I just followed that convention. However what the STREAM type is doing is slightly different, as it actually pays attention to the fileSystemOpts. I'd be happy to change it to just pass username and password!